core/src/main/scala/kafka/controller/PartitionStateMachine.scala

```diff
@@ -75,8 +75,7 @@ class PartitionStateMachine(controller: KafkaController) extends Logging {
   // register topic and partition change listeners
   def registerListeners() {
     registerTopicChangeListener()
-    if(controller.config.deleteTopicEnable)
-      registerDeleteTopicListener()
+    registerDeleteTopicListener()
```

Comment:
Should we deregister the delete topic listeners as well regardless of whether delete topic is enabled or not.